iothub/device/src/IotHubDeviceClient.cs

```diff
@@ -109,12 +103,7 @@ private IotHubDeviceClient(IotHubConnectionCredentials iotHubConnectionCredentia
             CancellationToken cancellationToken = default)
```

Comment:
Needs InvalidOperationException to be documented because of the OpenAsync check.